test/unit/common/Reducers.test.js

```diff
@@ -980,6 +981,21 @@ describe("Reducers", () => {
       });
       assert.deepEqual(state.spocs, INITIAL_STATE.DiscoveryStream.spocs);
     });
+    it("should add blocked spocs to blocked array with DISCOVERY_STREAM_SPOC_BLOCKED", () => {
+      const firstState = DiscoveryStream(undefined, {
+        type: at.DISCOVERY_STREAM_SPOC_BLOCKED,
+        data: { url: "https://foo.com" },
```

Comment:
what will happen if action.data is empty or null

Reply:
It does the same thing as `DISCOVERY_STREAM_LINK_BLOCKED`

They both assume action.data.url exists, iirc.

Reply:
To be more clear, if action.data is undefined, both events fail in the same way, when they try to access `action.data.url`, and hit a js error. If action.data is there, but no url, they both fail at doing their job but shouldn't throw a js error.

Both cases are currently unexpected to happen.

Actually, it would fail even sooner on `const filtered = spocsList.filter(s => s.url === action.data.url);` from `case at.PLACES_LINK_BLOCKED:`